packages/bot-web-ui/src/stores/download-store.js

```diff
@@ -0,0 +1,118 @@
+                localize('Reference ID (sell)'),
+                localize('Barrier'),
+                localize('Start Time'),
+                localize('Entry Spot'),
+                localize('Entry Spot Time'),
+                localize('Exit Spot'),
+                localize('Exit Spot Time'),
+                localize('Buy Price'),
+                localize('Profit/Loss'),
+            ],
+        ];
+        const transaction_messages = this.root_store.transactions.elements;
+        transaction_messages.map(item => {
+            if (item.type === 'contract') {
+                const arr = [
+                    item.data.display_name,
+                    item.data.transaction_ids.buy,
+                    item.data.transaction_ids.sell,
+                    item.data.barrier,
+                    item.data.date_start,
+                    item.data.entry_tick,
+                    item.data.entry_tick_time,
+                    item.data.exit_tick,
+                    item.data.exit_tick_time,
+                    item.data.buy_price,
+                    item.data.profit,
+                ];
+                transaction_csv_titles.push(arr);
+            }
+        });
+
+        // Push Transaction array to CSV
+        const transaction_csv_content = `data:text/csv;charset=utf-8, ${transaction_csv_titles
+            .map(e => e.join(','))
+            .join('\n')}`;
+        const transaction_encoded_uri = encodeURI(transaction_csv_content);
+        const transaction_link = document.createElement('a');
+        transaction_link.setAttribute('href', transaction_encoded_uri);
+        transaction_link.setAttribute('download', localize('Transactions.csv'));
+        document.body.appendChild(transaction_link);
+
+        transaction_link.click();
+    };
+
+    @action.bound
+    onClickDownloadJournal = () => {
+        // Journal Array
+        const journal_csv_titles = [[localize('Date'), localize('Time'), localize('Message')]];
+
+        const journal_messages = this.root_store.journal.filtered_messages;
+
+        journal_messages.map(item => {
+            let arrayMessage;
+            if (item.message_type !== 'success') {
+                arrayMessage = item.message;
+            } else {
+                arrayMessage = this.getSuccessJournalMessage(item.message.toString(), item.extra);
+            }
+
+            const arr = [item.date, item.time, arrayMessage];
+            journal_csv_titles.push(arr);
+        });
+
+        // Push Journal array to CSV
+        const journal_csv_content = `data:text/csv;charset=utf-8, ${journal_csv_titles
+            .map(e => e.join(','))
+            .join('\n')}`;
+        const journal_encoded_uri = encodeURI(journal_csv_content);
+        const journal_link = document.createElement('a');
+        journal_link.setAttribute('href', journal_encoded_uri);
+        journal_link.setAttribute('download', localize('Journal.csv'));
+        document.body.appendChild(journal_link);
```

Comment:
Should we remove these elements later on? We may end up filling the DOM with artifacts.

Reply:
Removed the element after download, thank you!